### PR TITLE
Recording player notes, giving points for correct notes, showing song results

### PR DESCRIPTION
**Pull request by achimmihca**
This is a major PR that implements the rest of the core gameplay loop. Now you can select a song, sing it, get points, the see results, select another song, sing it, etc.
The biggest problem with the gameplay is the microphone latency, which seems to be related to the way the Unity API handles microphone input. So I think we need a fast external lib to do this.

I did a major refactoring of the SingScene. The controlling in SingScene is now hierarchical.
The SingSceneController controls the non-player-related ui (e.g. TimeBar) as well as PlayerControllers. A PlayerController handles one player by interacting with more specialized scripts (e.g. recording notes, calculating scores, and updating the ui).
I advise you to open it in the unity editor and check the object hierarchy because this will explain the overall structure I think.

Furthermore, I implemented the SingingResultsScene with the design from USDX. I only did the ui for a single player, but it should be fairly easy to extend to more players and the logic for this is also already prepared.

I hope you did not implement anything big. This PR got much larger than I thought it would, sry.

### CODE REVIEW

**Comment by basisbit** [comment]
😅 regarding your last sentence... I only spent a few hours yesterday fixing a couple of code style things that Codacy complained about. Nothing big.
Suggestion: maybe the development would be more transparent if you use a feature branch in the UltraStar Play project. Only the master branch is strictly protected. You should already have all necessary contributor rights in this project. Using the feature branch would also add automatic codacy and travis-ci checks for each commit, thus resulting in faster feedback cycles. I'll try to update the travis-ci pipeline for unity 2019.1 today or on tuesday.
As always, this is just a suggestion. If you prefer working in your github-user-fork and create pull requests from there, that is also ok.

**Comment by achimmihca** [comment]
I see, so I will try a feature branch next time.

File: UltraStar Play/Assets/Common/Model/Song/Sentence/Note/ENoteType.cs [hunk 1]
```diff
@@ -2,6 +2,9 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+// TODO: What is Rap and RapGolden?
```
[the remark this answers is not in the document]
Comment (achimmihca) [reply UltraStar Play/Assets/Common/Model/Song/Sentence/Note/ENoteType.cs]: agreed

File: UltraStar Play/Assets/Common/Scene/SceneNavigator.cs [hunk 2]
```diff
@@ -34,7 +34,10 @@ public void AddSceneData(SceneData sceneData)
 
     public void LoadScene(EScene scene, SceneData sceneData)
     {
-        AddSceneData(sceneData);
+        if (sceneData != null)
```
[the remark this answers is not in the document]
Comment (achimmihca) [reply UltraStar Play/Assets/Common/Scene/SceneNavigator.cs]: agreed

File: UltraStar Play/Assets/Scenes/Sing/PlayerController/PlayerScoreController.cs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        get
+        {
+            return NormalNotesTotalScore + GoldenNotesTotalScore + PerfectSentenceBonusTotalScore;
+        }
+    }
+    public double NormalNotesTotalScore { get; private set; }
+    public double GoldenNotesTotalScore { get; private set; }
+    public double PerfectSentenceBonusTotalScore { get; private set; }
+
+    private double ScoreForACorrectBeatOfNormalNotes { get; set; }
```
[the remark this answers is not in the document]
Comment (achimmihca) [reply UltraStar Play/Assets/Scenes/Sing/PlayerController/PlayerScoreController.cs]: I removed the A

**Comment by basisbit** [comment]
By the way, you might want to take a look at the codacy analyais / comments. There are a bunch of helpful and good comments in there :)

**Pull request merged**